### ISSUE TITLE
Proposal: quantum classifiers

### ISSUE THREAD

**Issue by gcattan** [pyRiemann/pyRiemann#141]
Hello, 

We implemented a SKLearn wrapper around the Qiskit library to work with EEG data using Riemannian Geometry.

Could this work be considered as inside the scope of pyRiemann?

**Comment by qbarthelemy** [comment]
Could you share some specification about the API you want to add, and use-cases?
Are there publications on this topic?

**Comment by gcattan** [comment]
Litterature on the subject suggests it may offer an advantage as compared with classical computing for pattern recognition, computational time, and in situations where the size of the training sets is modest (e.g., [Blance et al. 2021](https://link.springer.com/article/10.1007/JHEP02(2021)212) and [Rebentrost et al, 2014](https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.113.130503)).
To my knowledge, there is no specific literature for the classification of EEG/MEG data using a quantum classifier, although we have an experiment undergoing.

Qiskit is an IBM library distributed under Apache 2.0 (https://qiskit.org/) which provides both quantum algorithms and infrastructure (simulator or real quantum computer, local or remote). The Github repository of the library is available at https://github.com/Qiskit.

We propose to use their implementation of support-vector machine and variational quantum classifiers [(Havlicek et al., 2018)](https://arxiv.org/pdf/1804.11326.pdf), which documentation is available [here](https://qiskit.org/documentation/stable/0.24/tutorials/machine_learning/03_vqc.html) and [there](https://qiskit.org/documentation/stable/0.24/tutorials/machine_learning/01_qsvm_classification.html)

A use case would be to use vectorized covariance matrices in the TangentSpace as an input for these classifiers, enabling a possible sandbox for researchers and engineers in the field. The implementation could be found [here](https://github.com/gcattan/pyRiemann/pull/2).

Another possible advantage of enabling quantum capabilities for pyRiemann (I can only suggest as I do not have expertise in this field), could be to take advantage of quantum optimization for convex problems:
https://arxiv.org/abs/1806.05343
https://arxiv.org/abs/1809.00643

**Comment by qbarthelemy** [comment]
Thank you for all these references!

What about creating a repo https://github.com/pyRiemann/pyRiemann-qiskit or https://github.com/pyRiemann/pyRiemann-contrib ? @agramfort @sylvchev

**Comment by sylvchev** [comment]
Seems like a good idea! I think an integration like that require to separate backend (Qiskit, or pytorch/tensorflow) from frontend API, and it is a long lasting job. It could help to have a working code available, in a pyriemann-qiskit repo.

**Comment by gcattan** [comment]
Hello, 
Thank you for your answers, this consideration about the design seems indeed important. 
I will be happy to contribute by raising a PR in the repository of your choice.

**Comment by qbarthelemy** [comment]
In your pull-request #139, tests will need to be updated, to follow the new tests deeply refactored in #136.

**Comment by gcattan** [comment]
Thanks for catching this. I updated the above PR and linked it to this issue. For the moment, I left the base branch untouched.

**Comment by gcattan** [comment]
Hum... there were some dependencies problems in the doc pipeline. It seemed easier to ignore it as the example will not be integrated into the official doc. Feel free to review when you have time :)

**Comment by qbarthelemy** [comment]
https://github.com/pyRiemann/pyRiemann-qiskit
Enjoy!

**Comment by gcattan** [comment]
@qbarthelemy thank you for the help. Please find the above proposals for the architecture and the implementation (two PR was raised for the Qiskit wrapper depending on whether or not you are willing to conserve the commit history).

**Comment by qbarthelemy** [comment]
Great! We can close this issue on pR side, and we will work on pRq side.
